### PR TITLE
Add Spark Conf to configuration

### PR DESCRIPTION
**Pull request by indig**
* Also remove userClasspathFirst spark config options from being applied
  by default
* minor refactoring of config
* minor bug fixes

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/config/package.scala [hunk 1]
```diff
@@ -45,4 +48,20 @@ package object config {
...
+      case (key, json) => json.fold(
+        Left("No null values allowed in this map"),
+        b => Right(key -> b.toString),
+        n => Right(key -> n.toString),
+        s => Right(key -> s),
+        _ => Left("No array values allowed in this map"),
+        _ => Left("No object values allowed in this map")
+      )
+    }.sequence.map(_.toMap)
+  }
```
Comment (indig) [inline polynote-kernel/src/main/scala/polynote/config/package.scala]: Thanks again for your help on this one @jeremyrsmith 😄

File: polynote-kernel/src/main/scala/polynote/messages/Messages.scala [hunk 2]
```diff
@@ -74,12 +74,15 @@ object NotebookCell {
 
 final case class NotebookConfig(
   dependencies: Option[DependencyConfigs],
-  repositories: Option[TinyList[RepositoryConfig]]
+  repositories: Option[TinyList[RepositoryConfig]],
+  sparkConfig: Option[ShortMap[String, String]]
```
Comment (indig) [inline polynote-kernel/src/main/scala/polynote/messages/Messages.scala]: Now that I look at it again, I wonder whether people are really going to need to add more than 256 values to their spark conf...

File: polynote-frontend/polynote/ui.js [hunk 3]
```diff
@@ -710,9 +710,10 @@ export class NotebookUI extends UIEventTarget {
         this.editBuffer = new EditBuffer();
 
         this.cellUI.addEventListener('UpdatedConfig', evt => {
+            console.log("updating config")
             const update = new messages.UpdateConfig(path, this.globalVersion, ++this.localVersion, evt.detail.config);
             this.editBuffer.push(this.localVersion, update);
-            this.socket.send(this.localVersion, update);
```
Comment (indig) [inline polynote-frontend/polynote/ui.js]: I think this was the only thing causing the Dependencies bug that @faisalzs ran into...

Comment (jeremyrsmith) [reply polynote-frontend/polynote/ui.js]: 😮 what kind of idiot.... *git blame*... er, I mean... good catch! 😅

File: polynote-frontend/polynote/ui.js [hunk 4]
```diff
@@ -710,9 +710,10 @@ export class NotebookUI extends UIEventTarget {
         this.editBuffer = new EditBuffer();
 
         this.cellUI.addEventListener('UpdatedConfig', evt => {
+            console.log("updating config")
```
Comment (jeremyrsmith) [inline polynote-frontend/polynote/ui.js]: We should do a purge of `console.log`s before alpha

File: polynote-kernel/src/main/scala/polynote/messages/Messages.scala (hunk 2 above)
Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/messages/Messages.scala]: It's just one byte, and this message won't be sent very often.

File: polynote-frontend/polynote/ui.js (hunk 4 above)
Comment (indig) [reply polynote-frontend/polynote/ui.js]: Woops, I thought I'd gotten all of 'em but I guess I missed that one!

**Pull request merged**